File: EduMS.Backend/src/EduMS.Domain/Entities/School.cs
using EduMS.Domain.Common;

namespace EduMS.Domain.Entities;

public class School : BaseAuditableEntity
{
    public string SchoolNameAr { get; set; } = string.Empty;
    public string SchoolNameEn { get; set; } = string.Empty;
    public string SchoolCode { get; set; } = string.Empty; // Unique identifier code
    public string Directorate { get; set; } = string.Empty; // Educational directorate
    public string Governorate { get; set; } = string.Empty; // Governorate
    public bool IsActive { get; set; } = true;
}

File: 04_Frontend_Application/Edums-Frontend/src/app/modules/m1-school-office/interface/school.ts
export interface School {
  id: number;
  directorateId: number | null;
  educationalStageId: number | null;
  schoolNameAr: string;
  schoolNameEn: string;
  schoolCode: string;
  directorate: string;
  governorate: string;
  establishmentDate: string | null;
  contactPhone: string | null;
  contactEmail: string | null;
  websiteUrl: string | null;
  postalAddress: string | null;
  taxRegistrationNumber: string | null;
  commercialLicenseNumber: string | null;
  maxStudentCapacity: number;
  isAccredited: boolean;
  isActive: boolean;
}

export type CreateSchoolDto = Omit<School, 'id' | 'isActive'>;

export type UpdateSchoolDto = Omit<School, 'isActive' | 'directorateId'>;
